cpp/mcap/include/mcap/mcap.hpp

```diff
@@ -696,6 +701,62 @@ struct TypedRecordReader {
   bool parsingChunk_;
 };
 
+struct LinearMessageView {
+  struct Iterator {
+    using iterator_category = std::forward_iterator_tag;
+    using difference_type = int64_t;
+    using value_type = Message;
+    using pointer = const Message*;
+    using reference = const Message&;
+
+    reference operator*() const;
+    pointer operator->();
+    Iterator& operator++();
+    Iterator operator++(int);
+    friend bool operator==(const Iterator& a, const Iterator& b);
+    friend bool operator!=(const Iterator& a, const Iterator& b);
+
+    static const Iterator& end() {
+      static LinearMessageView::Iterator emptyIterator;
+      return emptyIterator;
+    }
+
+  private:
+    friend LinearMessageView;
+
+    ProblemsList* problems_;
+    std::optional<TypedRecordReader> reader_;
+    Timestamp startTime_;
+    Timestamp endTime_;
+    Message curMessage_;
+
+    Iterator() = default;
+    Iterator(ProblemsList* problems, IReadable& dataSource, ByteOffset dataStart,
+             ByteOffset dataEnd, Timestamp startTime, Timestamp endTime);
+
+    void readNext();
+  };
+
+  LinearMessageView(const Status& status);
```

Comment:
prefer converting constructors to be `explicit` – but also, this seems like a somewhat strange constructor to offer... why not just `problems().push_back()`?